code/datums/elements/connect_loc.dm

```diff
@@ -54,6 +52,15 @@
 	if (!existing && isturf(tracked.loc))
 		RegisterSignal(tracked.loc, COMSIG_TURF_CHANGE, .proc/on_turf_change)
 
+/datum/element/connect_loc/proc/unregister_all(datum/listener)
+	for(var/atom/location as anything in targets)
+		var/list/loc_targets = targets[location]
+		for(var/atom/movable/tracked as anything in loc_targets)
+			if(loc_targets[tracked] != listener)
+				continue
+			unregister_signals(listener, tracked, location)
+			UnregisterSignal(tracked, COMSIG_MOVABLE_MOVED)
```

Comment:
Is this ok if there's more then one thing hooked into the listener?

Reply:
I was thinking about this a bit as I was working on this and I suspect there's probably a bug in here involving that, but this is how it worked before and I'd want to fix this issue separately if it is a problem.

Reply:
what do you mean by worked before? 
Oh also you may want to add a if(tracked) to that unregistersignal line, since it'll runtime for the valid location of null

Reply:
Look in Detach, the signal was unregistered from the tracked object and didn't do any checks there either. There probably is a problem there but it'll have to be looked at in more detail in another pr